target-riscv/cpu.c

```diff
@@ -96,16 +92,7 @@ static void riscv_cpu_disas_set_info(CPUState *s, disassemble_info *info) {
 static void riscv_cpu_realizefn(DeviceState *dev, Error **errp)
 {
     CPUState *cs = CPU(dev);
-    RISCVCPU *cpu = RISCV_CPU(dev);
     RISCVCPUClass *mcc = RISCV_CPU_GET_CLASS(dev);
-    CPURISCVState *env = &cpu->env;
-
-    /* Enable GC ISA */
-    set_feature(env, RISCV_FEATURE_RVM);
```

Comment:
In priv-1.9.1 the `misa` register is [writable](https://github.com/riscv/riscv-isa-manual/blob/master/src/machine.tex#L102-L103) and can be written to to disable instructions.  Writing a value to `misa` which clears the C bit will cause all compressed instructions to trap, etc.  This functionality is implemented in [spike](https://github.com/riscv/riscv-isa-sim/blob/master/riscv/processor.cc#L395) and [rocket](https://github.com/ucb-bar/rocket-chip/blob/master/src/main/scala/rocket/Rocket.scala#L203-L209) but was previously broken in qemu due to translation caching.  (It might still be broken, I haven't tested it.)